snuba/web/rpc/proto_component.py

```diff
@@ -0,0 +1,150 @@
+
+    @abstractmethod
+    def accept(self, visitor: ProtoVisitor) -> None:
+        pass
+
+
+class ColumnWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_ColumnWrapper(self)
+        column = self.underlying_proto
+        if column.HasField("formula"):
+            ColumnWrapper(column.formula.left).accept(visitor)
+            ColumnWrapper(column.formula.right).accept(visitor)
+
+
+class AggregationComparisonFilterWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_AggregationComparisonFilterWrapper(self)
+
+
+class AggregationFilterWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_AggregationFilterWrapper(self)
+        aggregation_filter = self.underlying_proto
+        if aggregation_filter.HasField("comparison_filter"):
+            AggregationComparisonFilterWrapper(
+                aggregation_filter.comparison_filter
+            ).accept(visitor)
+
+        if aggregation_filter.HasField("and_filter"):
+            for agg_filter in aggregation_filter.and_filter.filters:
+                AggregationFilterWrapper(agg_filter).accept(visitor)
+
+        if aggregation_filter.HasField("or_filter"):
+            for agg_filter in aggregation_filter.or_filter.filters:
+                AggregationFilterWrapper(agg_filter).accept(visitor)
+
+
+class TimeSeriesExpressionWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_TimeSeriesExpressionWrapper(self)
+
+        time_series_expression = self.underlying_proto
+        if time_series_expression.HasField("formula"):
+            TimeSeriesExpressionWrapper(time_series_expression.formula.left).accept(
+                visitor
+            )
+            TimeSeriesExpressionWrapper(time_series_expression.formula.right).accept(
+                visitor
+            )
+
+
+class TraceItemTableRequestWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_TraceItemTableRequestWrapper(self)
+        trace_item_table_request = self.underlying_proto
+        for col in trace_item_table_request.columns:
+            ColumnWrapper(col).accept(visitor)
+        for ob in trace_item_table_request.order_by:
+            ColumnWrapper(ob.column).accept(visitor)
+        if trace_item_table_request.HasField("aggregation_filter"):
+            AggregationFilterWrapper(
+                trace_item_table_request.aggregation_filter
+            ).accept(visitor)
+
+
+class TimeSeriesRequestWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_TimeSeriesRequestWrapper(self)
+        time_series_request = self.underlying_proto
+        for expression in time_series_request.expressions:
+            TimeSeriesExpressionWrapper(expression).accept(visitor)
+
+
+class ProtoVisitor(ABC):
```

Comment:
There's an important thing to note here in the comments for the user:

1. How do they use this class? You're doing some metaprogramming here, provide an example
2. With how this is implemented, it is important to understand that if the visitor traverses the structure, it will be traversed again by the wrapper object. The user should be given guidance to not traverse the object and have the visitor only work on the object it is visiting